server/tserver/src/main/java/org/apache/accumulo/tserver/metrics/CompactionExecutorsMetrics.java

```diff
@@ -24,115 +24,231 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
 import java.util.function.IntSupplier;
 import java.util.function.Supplier;
 
 import org.apache.accumulo.core.spi.compaction.CompactionExecutorId;
+import org.apache.accumulo.core.util.threads.ThreadPools;
 import org.apache.accumulo.tserver.compactions.CompactionManager.ExtCompMetric;
 import org.apache.hadoop.metrics2.lib.MetricsRegistry;
 import org.apache.hadoop.metrics2.lib.MutableGaugeInt;
 
 import com.google.common.collect.Sets;
 
+import io.micrometer.core.instrument.MeterRegistry;
+
 public class CompactionExecutorsMetrics extends TServerMetrics {
 
-  private volatile List<CeMetrics> ceml = List.of();
-  private Map<CompactionExecutorId,CeMetrics> metrics = new HashMap<>();
-  private Map<CompactionExecutorId,ExMetrics> exMetrics = new HashMap<>();
   private volatile Supplier<Collection<ExtCompMetric>> externalMetricsSupplier;
 
+  private volatile List<CeMetrics> ceMetricsList = List.of();
+  private final Map<CompactionExecutorId,CeMetrics> ceMetricsMap = new HashMap<>();
+  private final Map<CompactionExecutorId,ExMetrics> exCeMetricsMap = new HashMap<>();
+  private final MeterRegistry micrometerRegistry;
+
+  CompactionExecutorsMetricsHadoop hadoopMetrics;
+
   private static class CeMetrics {
-    MutableGaugeInt queuedGauge;
-    MutableGaugeInt runningGauge;
+    AtomicInteger queued;
+    AtomicInteger running;
 
     IntSupplier runningSupplier;
     IntSupplier queuedSupplier;
   }
 
   private static class ExMetrics {
-    MutableGaugeInt queuedGauge;
-    MutableGaugeInt runningGauge;
-
+    AtomicInteger queued;
+    AtomicInteger running;
   }
 
-  public CompactionExecutorsMetrics() {
+  public CompactionExecutorsMetrics(MeterRegistry registryMicrometer) {
     super("compactionExecutors");
+    this.micrometerRegistry = registryMicrometer;
+
+    ScheduledExecutorService scheduler =
+        ThreadPools.createScheduledExecutorService(1, "compactionExecutorsMetricsPoller", false);
+    Runtime.getRuntime().addShutdownHook(new Thread(scheduler::shutdownNow));
```

Comment:
Just like the `ThreadPools` class is for creating pools, there is a `Threads` class for a common way to create a Thread.